metrics/sources/kubelet/kubelet_client.go

```diff
@@ -130,33 +130,35 @@ type statsRequest struct {
 	Subcontainers bool `json:"subcontainers,omitempty"`
 }
 
-// Get stats for all non-Kubernetes containers.
-func (self *KubeletClient) GetAllRawContainers(host Host, start, end time.Time) ([]cadvisor.ContainerInfo, error) {
-	scheme := "http"
+func (self *KubeletClient) getScheme() string {
 	if self.config != nil && self.config.EnableHttps {
-		scheme = "https"
+		return "https"
+	} else {
+		return "http"
 	}
+}
 
+func (self *KubeletClient) getUrl(host Host, path string) string {
 	url := url.URL{
-		Scheme: scheme,
-		Host:   host.String(),
-		Path:   "/stats/container/",
+		Scheme: self.getScheme(),
+		Host:   fmt.Sprintf("%s:%d", host.IP, host.Port),
+		Path:   path,
 	}
 
-	return self.getAllContainers(url.String(), start, end)
+	return url.String()
```

Comment:
I'm wondering if it is preferable to return the URL object instead of the stirng here. All uses of this function at this point just use the String, so that's why I opted for returning the string instead of the object, but I'm on the fence.

Reply:
I'd return the URL itself.